ota-device-registry totally broken

Sorry for reporting here, but https://github.com/advancedtelematic/ota-device-registry/ isn't configured to allow bug reports. I just updated a test cluster from:

  advancedtelematic/device-registry:0.0.3-10-g883acfc

to:

 advancedtelematic/device-registry:0.3.0-11-g41be2f8

Its completed hosed my cluster. My devices show up as never seen online. If I create new devices, they can't talk to the service. Worse yet, nothing changes if I rollback. So it seems to completely corrupt my cluster.

FYI - I think I just reproduced this on master. I think I see the issue from the log here:
~~~
17:44:00,413 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
17:44:00,413 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
17:44:00,414 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [jar:file:/opt/docker/lib/com.advancedtelematic.device-registry-0.3.0-10-g8b86143.jar!/logback.xml]
17:44:00,427 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@6bb70ea0 - URL [jar:file:/opt/docker/lib/com.advancedtelematic.device-registry-0.3.0-10-g8b86143.jar!/logback.xml] is not of type file
17:44:00,537 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@53c57b50 - Adding [jar:file:/opt/docker/lib/com.advancedtelematic.libats-logging_2.12-0.2.1-2-g962e326.jar!/logback-libats.xml] to configuration watch list.
17:44:00,537 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@6bb70ea0 - URL [jar:file:/opt/docker/lib/com.advancedtelematic.libats-logging_2.12-0.2.1-2-g962e326.jar!/logback-libats.xml] is not of type file
17:44:00,541 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
17:44:00,547 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [stdout]
17:44:00,562 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
17:44:00,630 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
17:44:00,632 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [json]
17:44:00,702 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.classic.AsyncAppender]
17:44:00,707 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [async_json]
17:44:00,707 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [json] to ch.qos.logback.classic.AsyncAppender[async_json]
17:44:00,707 |-INFO in ch.qos.logback.classic.AsyncAppender[async_json] - Attaching appender named [json] to AsyncAppender.
17:44:00,709 |-INFO in ch.qos.logback.classic.AsyncAppender[async_json] - Setting discardingThreshold to 102
17:44:00,711 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.classic.AsyncAppender]
17:44:00,712 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [async_stdout]
17:44:00,712 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [stdout] to ch.qos.logback.classic.AsyncAppender[async_stdout]
17:44:00,712 |-INFO in ch.qos.logback.classic.AsyncAppender[async_stdout] - Attaching appender named [stdout] to AsyncAppender.
17:44:00,712 |-INFO in ch.qos.logback.classic.AsyncAppender[async_stdout] - Setting discardingThreshold to 102
17:44:00,714 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.zaxxer.hikari] to null, i.e. INHERITED
17:44:00,714 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [slick.backend.DatabaseComponent.action] to null, i.e. INHERITED
17:44:00,714 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.advancedtelematic] to null, i.e. INHERITED
17:44:00,715 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
17:44:00,716 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [stdout] to Logger[ROOT]
17:44:00,716 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
17:44:00,718 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@42398cd5 - Registering current configuration as safe fallback point
I|2019-04-09 17:44:00,756|akka.event.slf4j.Slf4jLogger|Slf4jLogger started
I|2019-04-09 17:44:00,950|c.a.ota.deviceregistry.Boot$|Running migrations
I|2019-04-09 17:44:00,973|o.f.c.internal.util.VersionPrinter|Flyway 4.0.3 by Boxfuse
I|2019-04-09 17:44:01,123|o.f.c.i.dbsupport.DbSupportFactory|Database: jdbc:mysql://mysql:3306/device_registry (MySQL 10.3)
I|2019-04-09 17:44:01,232|akka.event.slf4j.Slf4jLogger|Slf4jLogger started
I|2019-04-09 17:44:03,070|akka.event.slf4j.Slf4jLogger|Slf4jLogger started
I|2019-04-09 17:44:03,210|akka.event.slf4j.Slf4jLogger|Slf4jLogger started
I|2019-04-09 17:44:03,372|akka.event.slf4j.Slf4jLogger|Slf4jLogger started
I|2019-04-09 17:44:03,475|o.f.core.internal.command.DbValidate|Successfully validated 31 migrations (execution time 00:02.292s)
I|2019-04-09 17:44:03,541|o.f.core.internal.command.DbMigrate|Current version of schema `device_registry`: 23
I|2019-04-09 17:44:03,542|o.f.core.internal.command.DbMigrate|Migrating schema `device_registry` to version 24 - dynamic groups
I|2019-04-09 17:44:03,594|o.f.core.internal.command.DbMigrate|Migrating schema `device_registry` to version 25 - add created and modified to groups
I|2019-04-09 17:44:03,639|o.f.core.internal.command.DbMigrate|Migrating schema `device_registry` to version 26 - add indexed events
I|2019-04-09 17:44:03,688|o.f.core.internal.command.DbMigrate|Migrating schema `device_registry` to version 27 - device and ecu installation reports
I|2019-04-09 17:44:03,754|o.f.core.internal.command.DbMigrate|Migrating schema `device_registry` to version 28 - add indexed events archive
I|2019-04-09 17:44:03,788|o.f.core.internal.command.DbMigrate|Migrating schema `device_registry` to version 29 - add success flag to installation reports
I|2019-04-09 17:44:03,817|o.f.core.internal.command.DbMigrate|Migrating schema `device_registry` with repeatable migration AuditorInstallationReportsMigration
W|2019-04-09 17:44:03,861|slick.util.AsyncExecutor|Having maxConnection > maxThreads can result in deadlocks if transactions or database locks are used.
W|2019-04-09 17:44:06,936|a.h.impl.engine.client.PoolGateway|Connection attempt failed. Backing off new connection attempts for at least 100 milliseconds.
W|2019-04-09 17:44:07,091|a.h.impl.engine.client.PoolGateway|Connection attempt failed. Backing off new connection attempts for at least 200 milliseconds.
W|2019-04-09 17:44:07,421|a.h.impl.engine.client.PoolGateway|Connection attempt failed. Backing off new connection attempts for at least 400 milliseconds.
W|2019-04-09 17:44:07,930|a.h.impl.engine.client.PoolGateway|Connection attempt failed. Backing off new connection attempts for at least 800 milliseconds.
W|2019-04-09 17:44:08,822|a.h.impl.engine.client.PoolGateway|Connection attempt failed. Backing off new connection attempts for at least 1600 milliseconds.
W|2019-04-09 17:44:10,471|a.h.impl.engine.client.PoolGateway|Connection attempt failed. Backing off new connection attempts for at least 3200 milliseconds.
E|2019-04-09 17:44:10,477|o.f.core.internal.command.DbMigrate|Migration of schema `device_registry` with repeatable migration AuditorInstallationReportsMigration failed! Please restore backups and roll back database and code!
Exception in thread "main" java.lang.ExceptionInInitializerError
	at com.advancedtelematic.ota.deviceregistry.Boot.main(Boot.scala)
Caused by: org.flywaydb.core.api.FlywayException: Migration failed !
	at org.flywaydb.core.internal.resolver.jdbc.JdbcMigrationExecutor.execute(JdbcMigrationExecutor.java:47)
	at org.flywaydb.core.internal.command.DbMigrate.doMigrate(DbMigrate.java:352)
	at org.flywaydb.core.internal.command.DbMigrate.access$1100(DbMigrate.java:47)
	at org.flywaydb.core.internal.command.DbMigrate$4.doInTransaction(DbMigrate.java:308)
	at org.flywaydb.core.internal.util.jdbc.TransactionTemplate.execute(TransactionTemplate.java:72)
	at org.flywaydb.core.internal.command.DbMigrate.applyMigration(DbMigrate.java:305)
	at org.flywaydb.core.internal.command.DbMigrate.access$1000(DbMigrate.java:47)
	at org.flywaydb.core.internal.command.DbMigrate$2.doInTransaction(DbMigrate.java:230)
	at org.flywaydb.core.internal.command.DbMigrate$2.doInTransaction(DbMigrate.java:173)
	at org.flywaydb.core.internal.util.jdbc.TransactionTemplate.execute(TransactionTemplate.java:72)
	at org.flywaydb.core.internal.command.DbMigrate.migrate(DbMigrate.java:173)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:959)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:917)
	at org.flywaydb.core.Flyway.execute(Flyway.java:1373)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:917)
	at com.advancedtelematic.libats.slick.db.AsyncMigrations.$anonfun$migrate$1(BootMigrations.scala:32)
	at scala.runtime.java8.JFunction0$mcI$sp.apply(JFunction0$mcI$sp.java:12)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:654)
	at scala.util.Success.$anonfun$map$1(Try.scala:251)
	at scala.util.Success.map(Try.scala:209)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:288)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:91)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:81)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:91)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:44)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: akka.stream.StreamTcpException: Tcp command [Connect(localhost:8086,None,List(),Some(10 seconds),true)] failed because of java.net.ConnectException: Connection refused
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at akka.io.TcpOutgoingConnection$$anonfun$connecting$1.$anonfun$applyOrElse$3(TcpOutgoingConnection.scala:100)
	at akka.io.TcpOutgoingConnection.akka$io$TcpOutgoingConnection$$reportConnectFailure(TcpOutgoingConnection.scala:47)
	at akka.io.TcpOutgoingConnection$$anonfun$connecting$1.applyOrElse(TcpOutgoingConnection.scala:100)
	at akka.actor.Actor.aroundReceive(Actor.scala:517)
	at akka.actor.Actor.aroundReceive$(Actor.scala:515)
	at akka.io.TcpConnection.aroundReceive(TcpConnection.scala:31)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:592)
	at akka.actor.ActorCell.invoke(ActorCell.scala:561)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)
	at akka.dispatch.Mailbox.run(Mailbox.scala:225)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
~~~

Looks like the DB migration is dying because mysql was out of DB connections? I think I have a fix that I wasn't using in this deployment:
  https://github.com/foundriesio/ota-community-edition/commit/a061aaa56a603c8c666ad0cd16bc2e275b88301f

--- a/README.md
+++ b/README.md
@@ -4,6 +4,8 @@ Management of device status and configuration information. This project is part 
 
 ## Building
 
+You will have to set the environment variable `AUDITOR_MIGRATE=false` before running this service as part of the Community Edition.
+
 `sbt docker:publishLocal` - to publish docker image locally.  
 
 ## Running tests

--- a/src/main/resources/application.conf
+++ b/src/main/resources/application.conf
@@ -35,6 +35,8 @@ auditor = {
   port = 8086
   port = ${?AUDITOR_PORT}
   uri = "http://"${auditor.host}":"${auditor.port}
+  migrate = true
+  migrate = ${?AUDITOR_MIGRATE}
 }
 
 main {

--- a/src/main/scala/db/migration/R__AuditorInstallationReportsMigration.scala
+++ b/src/main/scala/db/migration/R__AuditorInstallationReportsMigration.scala
@@ -1,12 +1,13 @@
 package db.migration
+
 import akka.actor.ActorSystem
 import akka.stream.ActorMaterializer
 import com.advancedtelematic.libats.http.ServiceHttpClientSupport
 import com.advancedtelematic.libats.slick.db.AppMigration
 import com.advancedtelematic.ota.deviceregistry.client.AuditorHttpClient
 import com.advancedtelematic.ota.deviceregistry.db.MigrateOldInstallationReports
-import com.typesafe.config.ConfigFactory
-import slick.jdbc.MySQLProfile
+import com.typesafe.config.{Config, ConfigFactory}
+import slick.jdbc.MySQLProfile.api.Database
 
 import scala.concurrent.Future
 
@@ -16,9 +17,14 @@ class R__AuditorInstallationReportsMigration extends AppMigration with ServiceHt
   implicit val materializer: ActorMaterializer = ActorMaterializer()
   import system.dispatcher
 
-  private val auditorUri = ConfigFactory.load().getString("auditor.uri")
+  private val config: Config = ConfigFactory.load()
+  private val migrateAuditor = config.getBoolean("auditor.migrate")
+  private val auditorUri = config.getString("auditor.uri")
   private val auditor = new AuditorHttpClient(auditorUri, defaultHttpClient)
 
-  override def migrate(implicit db: MySQLProfile.api.Database): Future[Unit] =
-    new MigrateOldInstallationReports(auditor).run.map(_ => ())
+  override def migrate(implicit db: Database): Future[Unit] =
+    if (migrateAuditor)
+      new MigrateOldInstallationReports(auditor).run.map(_ => ())
+    else
+      Future.unit
 }

